carbonmark/components/ChangeLanguageButton/index.tsx

```diff
@@ -0,0 +1,68 @@
+import { t } from "@lingui/macro";
+import Language from "@mui/icons-material/Language";
+import Tippy from "@tippyjs/react";
+import { activateLocale, loadTranslation, locales } from "lib/i18n";
+import { FC, useEffect, useState } from "react";
+import * as styles from "./styles";
+
+export const ChangeLanguageButton: FC = () => {
+  const [showMenu, setShowMenu] = useState(false);
+  const [locale, setLocale] = useState("en");
+
+  useEffect(() => {
+    const locale = window?.localStorage?.getItem("locale") || "en";
+    setLocale(locale);
+    selectLocale(locale);
+  }, []);
+
+  const selectLocale = async (locale: string) => {
+    const messages = await loadTranslation(locale);
+    await activateLocale(locale, messages);
+    window?.localStorage?.setItem("locale", locale);
+    setLocale(locale);
+  };
```

Comment:
This is the approach one would take for a single page app where locales are managed client-side (like our `app` package), but Carbonmark can use nextjs builtins. See `site` for an example:

https://github.com/KlimaDAO/klimadao/blob/9d32d9678e53dea2566fd45ca3b61b4283fc6803/site/components/ChangeLanguageButton/index.tsx#L13